lib/core/app/controllers/concerns/ros/application_controller_concern.rb

```diff
@@ -13,45 +13,23 @@ module ApplicationControllerConcern
       after_action :set_headers!
 
       def authenticate_it!
+        Ros::Sdk::Credential.authorization = request.env['HTTP_AUTHORIZATION']
         return unless (@current_user = request.env['warden'].authenticate!(:api_token))
 
-        if auth_type.basic?
+        if auth_type.eql?('basic')
           @current_jwt = Jwt.new(current_user.jwt_payload)
-        elsif auth_type.bearer?
+        elsif auth_type.eql?('bearer')
           @current_jwt = Jwt.new(request.env['HTTP_AUTHORIZATION'])
           if (sub_cognito = current_jwt.claims['sub_cognito'])
             @cognito_user_urn = Ros::Urn.from_urn(sub_cognito)
             @cognito_user_id = cognito_user_urn.resource_id
           end
         end
+        # TODO: Credential.authorization must be an instance variable
+        current_jwt.add_claims(user: current_user.to_json) unless current_jwt.claims.key?('user')
```

Comment:
```claims``` is now a ```HashWithIndifferentAccess``` so you don't have to use strings.

Reply:
that's not the case when the jwt has been decoded by the JWT library. it is a string based hash unfortunately

Reply:
I misuderstood. yes, you are correct. tnx